### PR TITLE
feat(ui): accessible collapsible component

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
We've added a clear outline to the summary of the collapsible components. 

closes: [HRZ-90969](https://spryker.atlassian.net/browse/HRZ-90969)

[HRZ-90969]: https://spryker.atlassian.net/browse/HRZ-90969?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/base/ui/structure/collapsible/src/collapsible.component.ts [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
+import { collapsibleStyles } from './collapsible.styles';
 import { collapsibleBaseStyle } from './styles';
 
 export class CollapsibleComponent
   extends I18nMixin(LitElement)
   implements CollapsibleAttributes
 {
-  static styles = [collapsibleBaseStyle];
+  static styles =
+    featureVersion >= '1.4' ? collapsibleStyles : [collapsibleBaseStyle];
```
Comment (Sinucid) [inline libs/base/ui/structure/collapsible/src/collapsible.component.ts:19]: Is this a new convention?

File: libs/base/ui/structure/collapsible/src/styles/themes/storefront.styles.ts [hunk 2]
```diff
@@ -34,10 +34,9 @@ const blockAppearance = css`
     box-shadow: var(--oryx-elevation-0) var(--oryx-color-elevation);
   }
 
-  :host(${blockSelector}:active),
-  :host(${blockSelector}[open]) {
+  /* :host(${blockSelector}:active) {
```
Comment (Sinucid) [inline libs/base/ui/structure/collapsible/src/styles/themes/storefront.styles.ts]: left over?

File: libs/base/ui/structure/collapsible/src/styles/themes/storefront.styles.ts [hunk 3]
```diff
@@ -47,10 +46,19 @@ const blockAppearance = css`
...
   */
-  summary:focus-visible::after {
-    box-shadow: var(--oryx-box-shadow-focus);
+
+  summary:focus-visible {
+    outline: solid 1px var(--oryx-color-focus);
   }
 
+  :host([open]) summary:focus-visible {
+    /* outline-offset: -4px; */
```
Comment (Sinucid) [inline libs/base/ui/structure/collapsible/src/styles/themes/storefront.styles.ts]: The same question

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/base/ui/form/checkbox/src/checkbox.styles.ts [hunk 4]
```diff
@@ -1,5 +1,18 @@
...
+const focusStyles =
+  featureVersion >= '1.4'
+    ? css`
+        ::slotted(input:focus-visible)::after {
+          outline: solid 1px var(--oryx-color-focus);
+          outline-offset: 4px;
+          border-radius: 1px;
+        }
+      `
+    : (`` as any);
```
Comment (Sinucid) [inline libs/base/ui/form/checkbox/src/checkbox.styles.ts]:
```suggestion
    :  css``;
```

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/base/ui/structure/collapsible/src/collapsible.component.ts (hunk 1 above)
Comment (tobi-or-not-tobi) [reply libs/base/ui/structure/collapsible/src/collapsible.component.ts:19]: kind of, but not yet formalised. Few things that cross my mind when making changes For this PR:
- the way we've implemented themes is not so convenient. Say you introduce a new theme, you'd loose all styles for components that are theme driven (e.g. storefront vs backoffice). Going forward, I like our themes to only influence CSS variables. We could still add styles per theme, but the basic components should work properly. 
- keep it simple. I rather have single css file per component, and not an array. Complex CSS is likely an indicator  that the complex is too large anyway.

File: libs/base/ui/form/checkbox/src/checkbox.styles.ts (hunk 4 above)
Comment (tobi-or-not-tobi) [reply libs/base/ui/form/checkbox/src/checkbox.styles.ts]: won't work as it throws `Unexpected empty source  no-empty-source` in stylelint. This only works if it holds style rules. I've tried many solutions and ended up with this one.

**Review by Sinucid — DISMISSED** [review]

**Review by Sinucid — APPROVED** [review]

**Pull request merged**